### ISSUE TITLE
[Feature]: Support custom args in OpenAI (chat) completion requests

### ISSUE THREAD

**Issue by njhill** [vllm-project/vllm#16802]
https://github.com/vllm-project/vllm/pull/13300 added an `extra_args` field to `SamplingParameters` to allow for passing arbitrary parameters through to custom sampler / logits processor implementations.

A follow on is to expose this in the OpenAI API endpoints, see discussion at the end of that PR.

This will be needed to support custom sampling parameters via V1 logits processor plugins once that's ready (https://github.com/vllm-project/vllm/pull/13360).

**Comment by abf149** [comment]
Picking up this workstream FYI.

**Comment by afeldman-nm** [comment]
@WoosukKwon @comaniac (or anyone else), interested in your thoughts on what is the best manner in which to integrate custom args into the API.

For example I proposed a solution based on adding an `extra_args` field

```
# Custom args param
extra_args: Optional[dict[str, Any]] = Field(
    default=None,
    description=("Additional kwargs to pass to sampling."),
)
```

to `CompletionRequest`, `ChatCompletionRequest`, and `TranscriptionRequest`, which can be found here

https://github.com/vllm-project/vllm/pull/16862

But I am interested in your thoughts on whether this is the best way to proceed?
